### DISCUSSION TITLE
Keyboard shortcut restrictions?

### DISCUSSION THREAD

**Discussion by PatrickF1** [rxhanson/Rectangle#314] [Q&A]
What does the preference "Remove keyboard shortcut restrictions" control?

**Comment by rxhanson** [comment]
There are some restrictions that the keyboard shortcut recorder has by default for making sure you don't override shortcuts that are in the application's menu or in macOS. It's basically just safeguards, and Rectangle has the "default" set of menu items so that the shortcut recorder will catch those and let you know if you're overwriting commonly used menu item shortcuts, like copy/paste. The toggle is for removing those safeguards. I had a lot of users that didn't care that they were overriding shortcuts that were being blocked by the shortcut recorder, hence the toggle.

This used to be "Allow any keyboard shortcut", but that didn't make sense since you're still restricted from overwriting shortcuts globally set by other apps, and you can't technically record any shortcut you want. 

I think I will change this to "Remove keyboard shortcut recorder safeguards". Let me know if you can think of anything that would be more straightforward than that.

**Reply by PatrickF1** [reply]
Thank you for that clear explanation! I think more importantly than renaming the button is including a description in the documentation.

**Reply by ronin712** [reply]
I agree that "Remove keyboard shortcut recorder safeguards" is more descriptive.  But whether you change the label or not, I think the most helpful thing would be to add an information popup (like you have on "Show Todo Mode in menu") that describes it the way you did in this comment.
